xdsl/ir/core.py

```diff
@@ -1660,6 +1660,39 @@ class Block(_IRNode, IRWithUses):
     parent: Region | None = field(default=None, repr=False)
     """Parent region containing the block."""
 
+    _name: str | None = field(init=False, default=None)
+
+    _name_regex: ClassVar[re.Pattern[str]] = re.compile(r"([A-Za-z_$.-][\w$.-]*)")
+
+    @staticmethod
+    def _is_default_block_name(name: str) -> bool:
+        """Check if a name matches the default block naming pattern (bb followed by digits)."""
+        return name.startswith("bb") and name[2:].isdigit() and name[2:] != ""
+
+    @property
+    def name_hint(self) -> str | None:
+        return self._name
+
+    @name_hint.setter
+    def name_hint(self, name: str | None):
+        # only allow valid names
+        if Block.is_valid_name(name):
+            # Remove `_` followed by numbers at the end of the name
+            if name is not None:
+                r1 = re.compile(r"(_\d+)+$")
```

Comment:
This line makes me sad in SSAValue, can you please keep sadness levels constant by only compiling the regex once, or lowering them by making a similar change for SSAValue?

Reply:
You turned my frown upside down